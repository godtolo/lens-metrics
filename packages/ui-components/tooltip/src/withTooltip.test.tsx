/**
 * Copyright (c) OpenLens Authors. All rights reserved.
 * Licensed under MIT License. See LICENSE in root directory for more information.
 */

import type { SingleOrMany } from "@k8slens/utilities";
import { render } from "@testing-library/react";
import React from "react";
import { withTooltip } from "./withTooltip";

type MyComponentProps = {
  text: string;
  id?: string;
  children?: SingleOrMany<React.ReactNode>;
};

const MyComponent = withTooltip(({ text }: MyComponentProps) => <div>{text}</div>);

describe("withTooltip tests", () => {
  it("does not render a tooltip when not specified", () => {
    const result = render(<MyComponent text="foobar" />);

    expect(result.baseElement).toMatchSnapshot();
  });

  it("renders a tooltip when specified", () => {
    const result = render(<MyComponent text="foobar" tooltip="my-tooltip" id="bat" />);

    expect(result.baseElement).toMatchSnapshot();
  });

  it("renders a tooltip when specified via tooltip props", () => {
    const result = render(
      <MyComponent text="foobar" tooltip={{ children: "my-tooltip" }} id="bat" />,
    );

    expect(result.baseElement).toMatchSnapshot();
  });

  it("renders a tooltip when specified without id", () => {
    const result = render(<MyComponent text="foobar" tooltip="my-tooltip" />);

    expect(result.baseElement).toMatchSnapshot();
  });
});
