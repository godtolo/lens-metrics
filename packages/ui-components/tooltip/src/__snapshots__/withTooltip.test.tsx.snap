// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withTooltip tests does not render a tooltip when not specified 1`] = `
<body>
  <div>
    <div>
      foobar
    </div>
  </div>
</body>
`;

exports[`withTooltip tests renders a tooltip when specified 1`] = `
<body>
  <div>
    <div>
      foobar
    </div>
  </div>
</body>
`;

exports[`withTooltip tests renders a tooltip when specified via tooltip props 1`] = `
<body>
  <div>
    <div>
      foobar
    </div>
  </div>
</body>
`;

exports[`withTooltip tests renders a tooltip when specified without id 1`] = `
<body>
  <div>
    <div>
      foobar
    </div>
  </div>
</body>
`;
